core/templates/pages/admin-page/misc-tab/admin-misc-tab.directive.ts

```diff
@@ -86,16 +80,15 @@ angular.module('oppia').directive('adminMiscTab', [
             return;
           }
           ctrl.regenerationMessage = 'Regenerating opportunities...';
-          $http.post(ADMIN_HANDLER_URL, {
-            action: 'regenerate_topic_related_opportunities',
-            topic_id: ctrl.topicIdForRegeneratingOpportunities
-          }).then(function(response) {
+          AdminMiscTabBackendApiService.regenerateTopicRelatedOpportunities
+          (ctrl.topicIdForRegeneratingOpportunities)
+            .then(function(response) {
             ctrl.regenerationMessage = (
               'No. of opportunities model created: ' +
-              response.data.opportunities_count);
+              response.opportunities_count);
           }, function(errorResponse) {
             ctrl.regenerationMessage = (
-              'Server error: ' + errorResponse.data.error);
+              'Server error: ' + errorResponse.error.error);
```

Comment:
Ditto as above* (here and elsswhere)

Reply:
Same as above mentioned, If we use data code is breaking.